### PR TITLE
Reorganize tests

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: tests/Common/Reader/ReaderTrait.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+namespace Yiisoft\Data\Tests\Common\Reader;
+
+use Yiisoft\Data\Reader\DataReaderInterface;
+use Yiisoft\Data\Reader\Iterable\IterableDataReader;
+
+trait ReaderTrait
+{
+    protected function getReader(): DataReaderInterface
+    {
+        return new IterableDataReader(self::$fixtures);
```
Comment (vjik) [inline tests/Common/Reader/ReaderTrait.php:14]: `IterableDataReader` is not common.

File: tests/Common/FixtureTrait.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Data\Tests\Common;
+
+trait FixtureTrait
```
Comment (vjik) [inline tests/Common/FixtureTrait.php:7]: Do this trait need somewhere else besides readers?

File: tests/Common/Reader/FilterHandler/AllHandlerWithReaderTestTrait.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+use Yiisoft\Data\Reader\Filter\All;
+use Yiisoft\Data\Reader\Filter\Equals;
+
+trait AllHandlerWithReaderTestTrait
+{
+    public function testWithReader(): void
+    {
+        $reader = $this
+            ->getReader()
```
Comment (vjik) [inline tests/Common/Reader/FilterHandler/AllHandlerWithReaderTestTrait.php:15]: Suggestion only...

Create abstract classes `Base%Filter%WithReaderTest` with abstract method `getReader()` and use `FixtureTrait`.

File: tests/Common/Reader/FilterHandler/ILikeHandlerWithReaderTestTrait.php
Comment (vjik) [inline tests/Common/Reader/FilterHandler/ILikeHandlerWithReaderTestTrait.php:1]: Not need

**Review by samdark — APPROVED** [review]

**Review by vjik — APPROVED** [review]

**Pull request merged**